pages/en/cookbook/migrating-a-subgraph.mdx

```diff
@@ -63,17 +65,74 @@ graph codegen && graph build
 
 - Remember that publishing is an on-chain action and will require gas to be paid for in Ethereum - see an example transaction [here](https://etherscan.io/tx/0xd0c3fa0bc035703c9ba1ce40c1862559b9c5b6ea1198b3320871d535aa0de87b). Prices are roughly around 0.0425 ETH at 100 gwei.
 - Any time you need to upgrade your subgraph, you will be charged an upgrade fee. Remember, upgrading is just publishing another version of your existing subgraph on-chain. Because this incurs a cost, it is highly recommended to deploy and test your subgraph on Goerli before deploying to mainnet. It can, in some cases, also require some GRT if there is no signal on that subgraph. In the case there is signal/curation on that subgraph version (using auto-migrate), the taxes will be split.
 
-And that's it! After you are done publishing, you'll be able to view your subgraphs live on the network via [The Graph Explorer](https://thegraph.com/explorer).
+7. Publish the subgraph on The Graph's decentralized network by hitting the "Publish" button.
+
+And that's it! After you are done publishing, you'll be able to view your subgraphs live on the decentralized network via [The Graph Explorer](https://thegraph.com/explorer).
 
 Feel free to leverage the [#Curators channel](https://discord.gg/rC8rBuRtbH) on Discord to let Curators know that your subgraph is ready to be signaled. It would also be helpful if you share your expected query volume with them. Therefore, they can estimate how much GRT they should signal on your subgraph.
 
-### Upgrading a Subgraph on the Network
+### Create an API key
+
+You can generate an API key in Subgraph Studio [here](https://thegraph.com/studio/apikeys/).
+
+![API key creation page](/img/api-image.png)
+
+At the end of each week, an invoice will be generated based on the query fees that have been incurred during this period. This invoice will be paid automatically using the GRT available in your balance. Your balance will be updated after the cost of your query fees are withdrawn. Query fees are paid in GRT via the Polygon network. You will need to add GRT to the Polygon billing contract to enable your API key via the following steps:
+
+- Purchase GRT on an exchange of your choice
+- Send the GRT to your wallet
+- On the Billing page in Studio, click on Add GRT
+
+![Add GRT in billing](/img/add-grt-in-billing.png)
+
+- Follow the steps to bridge the GRT to Polygon.
+- Once bridged, add the GRT to your billing balance
+
+![Billing pane](/img/billing-pane.png)
+
+> Note: This process is slightly different if you are using a multisig since you can not bridge GRT to Polygon with a multisig. You would need to send the GRT to an EOA (externally owned address) and bridge it from there to Polygon. Once on Polygon, you can load the multisig address. We have created a special tool to help you through these steps: https://multisig-billing.thegraph.com/
```

Comment:
```suggestion
> Note: This process is slightly different if you are using a multisig since you can not bridge GRT to Polygon with a multisig. You would need to send the GRT to an EOA (externally owned address) and bridge it from there to Polygon. Once on Polygon, you can load the multisig address. We have created [a special tool to help you through these steps](https://multisig-billing.thegraph.com/).
```